src/comment_parser.rs

```diff
@@ -505,3 +533,53 @@ impl<'a> Iterator for ParentheticalIter<'a> {
         Some((self.to_span(all), item))
     }
 }
+
+/// A date, as understood by metamath tools.
+/// This is just a `dd-mmm-yyyy` field after parsing,
+/// so it has weak calendrical restrictions.
+#[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq)]
+pub struct Date {
+    /// A year, which must be a 4 digit number (0-9999).
+    pub year: u32,
+    /// A month, parsed from three letter names: `Jan`, `Feb`, etc. (1-12)
+    pub month: u8,
+    /// A day, parsed from a 1 or 2 digit number (0-99).
+    pub day: u8,
+}
```

Comment:
Yes, this is very deliberately not using a real calendar, because that is liable to cause locale- and time-specific issues. Rather, I would like dates to be treated as a simple collection of numbers with clear roundtrip rules and a linear order. The only thing I might change later is to require that the day be in the 1-31 range instead of 0-99. For generating new contribution comments, it makes sense to use the actual date on the calendar to initialize a `Date` struct, but that is only when authoring a new proof; for handling old proofs we can't say much about the dates in the file.

Reply:
Anyway if needed conversion to/from any other date format can be done outside of the library, and if we see the need we can still propose conversions in the future.